src/org/centos/pipeline/PipelineUtils.groovy

```diff
@@ -113,51 +126,273 @@ def checkLastImage(stage) {
+                "compose_rev=${env.commit}\n" +
                 "image_url=''\n" +
                 "image_name=''\n" +
                 "type=qcow2\n"
     } else if ((messageType == 'image.complete') || (messageType == 'test.smoke.running') ||
             (messageType == 'test.smoke.compelete')) {
         messageProperties = messageProperties +
-                "compose_url=${HTTP_BASE}/artifacts/${RSYNC_DIR}/${branch}/ostree\n"
-                "compose_rev=${commit}\n" +
-                "image_url=${image2boot}\n" +
-                "image_name=${image_name}\n" +
+                "compose_url=${env.HTTP_BASE}/artifacts/${env.RSYNC_DIR}/${env.branch}/ostree\n"
+                "compose_rev=${env.commit}\n" +
+                "image_url=${env.image2boot}\n" +
+                "image_name=${env.image_name}\n" +
                 "type=qcow2\n"
     } else {
         return [ topic, messageProperties, messageContent ]
     }
     return [ topic, messageProperties, messageContent ]
+}
+
+/**
+ * Library to send message
+ *
+ * variables
+ *  msgProps - The message properties
+ *  msgContent - The content of the message
+ */
+def sendMessage(msgProps, msgContent) {
+    sendCIMessage messageContent: msgContent,
+            messageProperties: msgProps,
+            messageType: 'Custom',
+            overrides: [topic: "${topic}"],
+            providerName: "${MSG_PROVIDER}"
+}
+
+/**
+ * Library to parse CI_MESSAGE and inject its key/value pairs as env variables.
+ *
+ */
+def injectFedmsgVars() {
+
+    // Parse the CI_MESSAGE into a Map
+    def ci_data = new JsonSlurper().parseText(env.CI_MESSAGE)
+
+    // If we have a 'commit' key in the CI_MESSAGE, for each key under 'commit', we
+    // * prepend the key name with fed_
+    // * replace any '-' with '_'
+    // * truncate the value for the key at the first '\n' character
+    // * replace any double-quote characters with single-quote characters in the value for the key.
+
+    if (ci_data['commit']) {
+        ci_data.commit.each { key, value ->
+            env."fed_${key.toString().replaceAll('-', '_')}" =
+                    value.toString().split('\n')[0].replaceAll('"', '\'')
+        }
+        if (env.fed_branch == 'master'){
+            env.branch = 'rawhide'
+        } else {
+            env.branch = env.fedbranch
+        }
+    }
+}
+
+/**
+ * Library to set default environmental variables. Performed once at start of Jenkinsfile
+ * variables
+ *  envMap - A map of key/value pairs which will be set as environmental variables.
+ */
+def setDefaultEnvVars(envMap=null){
+    env.MAIN_TOPIC = env.MAIN_TOPIC ?: 'org.centos.prod'
+    env.MSG_PROVIDER = env.MSG_PROVIDER ?: 'fedora-fedmsg'
+    env.HTTP_BASE = env.HTTP_BASE ?: 'http://artifacts.ci.centos.org/artifacts/fedora-atomic'
+    env.RSYNC_USER = env.RSYNC_USER ?: 'fedora-atomic'
+    env.RSYNC_SERVER = env.RSYNC_SERVER ?: 'artifacts.ci.centos.org'
```

Comment:
You need RSYNC_DIR here and should be set to stage if the env.ghprbActualCommit is a sha

Reply:
Per our conversation, I thought we were setting env.RSYNC_DIR based on whether env.MAIN_TOPIC containg "staging"?